src/main/resources/assets/computercraft/lua/rom/autorun/10_require.lua

```diff
@@ -0,0 +1,124 @@
+-- -*- coding: utf8 -*-
+-- Copyright (c) 2014 Odd Straaboe <oddstr13 at openshell dot no>
+-- License: MIT - http://opensource.org/licenses/MIT
+-- Filename: require.lua
+
+_G.package = {}
+
+_G.package.cpath = ""
+_G.package.loaded = {}
+_G.package.loadlib = function() error("NotImplemented: package.loadlib") end
+_G.package.path = table.concat({
+    "?",
+    "?.lua",
+    "?/init.lua",
+    "/lib/?",
+    "/lib/?.lua",
+    "/lib/?/init.lua",
+    "/rom/apis/?",
+    "/rom/apis/?.lua",
+    "/rom/apis/?/init.lua",
+    "/rom/apis/turtle/?",
+    "/rom/apis/turtle/?.lua",
+    "/rom/apis/turtle/?/init.lua",
+    "/rom/apis/command/?",
+    "/rom/apis/command/?.lua",
+    "/rom/apis/command/?/init.lua",
+}, ";")
+_G.package.preload = {}
+_G.package.seeall = function(module) error("NotImplemented: package.seeall") end
+_G.module = function(m) error("NotImplemented: module") end
+
+local _package_path_loader = function(name)
+    
+    local fname = name:gsub("%.", "/")
+    
+    for pattern in package.path:gmatch("[^;]+") do
+        
+        local fpath = pattern:gsub("%?", fname)
+        
+        if fs.exists(fpath) and not fs.isDir(fpath) then
+            
+            local apienv = {}
+            setmetatable(apienv, {__index = _G})
+            
+            local apifunc, err = loadfile(fpath)
+            local ok
+            
+            if apifunc then
+                setfenv(apifunc, apienv)
+                ok, err = pcall(apifunc)
+            end
+            
+            if not apifunc or not ok then
+                error("error loading module '" .. name .. "' from file '" .. fpath .. "'\n\t" .. err)
+            end
+            
+            local api = {}
+            if type(err) == "table" then
+              api = err
+            end
+            for k,v in pairs( apienv ) do
+                api[k] = v
+            end
+            
+            return api
+        end
+    end
+end
+
+_G.package.loaders = {
+    function(name)
+        if package.preload[name] then
+            return package.preload[name]
+        else
+            return "\tno field package.preload['" .. name .. "']"
+        end
+    end,
+    
+    function(name)
+        local _errors = {}
+        
+        local fname = name:gsub("%.", "/")
+        
+        for pattern in package.path:gmatch("[^;]+") do
+            
+            local fpath = pattern:gsub("%?", fname)
+            if fs.exists(fpath) and not fs.isDir(fpath) then
+                return _package_path_loader
+            else
+                table.insert(_errors, "\tno file '" .. fpath .. "'")
+            end
+        end
+        
+        return table.concat(_errors, "\n")
+    end
+}
+
+_G.require = function(name)
+    if package.loaded[name] then
+        return package.loaded[name]
+    end
```

Comment:
Ideally we could check for package loops here (a requires b, b requires a). Lua does this by storing a magic value in `package.loaded` before loading it, and checking it is equal to that here. Note that this check will also fail if a package returns `false`.